app/src/editor/shared/components/modules/Scheduler.jsx

```diff
@@ -0,0 +1,714 @@
+            onChange={(value) => onChange({
+                endDate: {
+                    month: value,
+                    day: endDate.day,
+                    hour: endDate.hour,
+                    minute: endDate.minute,
+                },
+            })}
+        />
+        at <HourSelect
+            value={endDate.hour}
+            onChange={(value) => onChange({
+                endDate: {
+                    month: endDate.month,
+                    day: endDate.day,
+                    hour: value,
+                    minute: endDate.minute,
+                },
+            })}
+        />:<MinuteSelect
+            value={endDate.minute}
+            onChange={(value) => onChange({
+                endDate: {
+                    month: endDate.month,
+                    day: endDate.day,
+                    hour: endDate.hour,
+                    minute: value,
+                },
+            })}
+        />
+    </Fragment>
+)
+
+const defaultDates = [
+    { // hour
+        startDate: { minute: 0 },
+        endDate: { minute: 0 },
+    },
+    { // day
+        startDate: {
+            hour: 0,
+            minute: 0,
+        },
+        endDate: {
+            hour: 0,
+            minute: 0,
+        },
+    },
+    { // week
+        startDate: {
+            weekday: 1,
+            hour: 0,
+            minute: 0,
+        },
+        endDate: {
+            weekday: 1,
+            hour: 0,
+            minute: 0,
+        },
+    },
+    { // month
+        startDate: {
+            day: 1,
+            hour: 0,
+            minute: 0,
+        },
+        endDate: {
+            day: 1,
+            hour: 0,
+            minute: 0,
+        },
+    },
+    { // year
+        startDate: {
+            month: 1,
+            day: 1,
+            hour: 0,
+            minute: 0,
+        },
+        endDate: {
+            month: 1,
+            day: 1,
+            hour: 0,
+            minute: 0,
+        },
+    },
+]
+
+export class RuleComponent extends React.Component {
+    onRemove = () => {
+        const { onRemove, rule } = this.props
+
+        if (onRemove) {
+            onRemove(rule.id)
+        }
+    }
+
+    onIntervalChange = (intervalType) => {
+        this.props.onChange(this.props.rule.id, {
+            intervalType,
+            ...(defaultDates[intervalType] || {}),
+        })
+    }
+
+    onUpdateDates = (properties) => {
+        this.props.onChange(this.props.rule.id, {
+            ...properties,
+        })
+    }
+
+    onValueChange = (value) => {
+        this.props.onChange(this.props.rule.id, {
+            value,
+        })
+    }
+
+    render() {
+        const { isHovered, rule, isActive } = this.props
+
+        return (
+            <div className={styles.ruleContainer}>
+                {isHovered && (
+                    <div // eslint-disable-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
```

Comment:
To silence eslint could this be an actual button instead of a clickable div maybe?